File: CS/GridControlCellTemplate/Model/Item.cs
using DevExpress.Mvvm;
using System;
using System.Collections.Generic;
using System.IO;
using System.Text.RegularExpressions;

namespace GridControlCellTemplate.Model
{
    public class Record : BindableBase
    {
        public string Text {
            get => GetValue<string>();
            set => SetValue(value);
        }

        public bool IsRead {
            get => GetValue<bool>();
            set => SetValue(value);
        }

        public Record(string text) {
            Text = text;
        }

        public static IEnumerable<Record> GetData(int quantity) {
            var sentences = Regex.Split(File.ReadAllText(@"Model/Sentences.txt"), @"(?<=[\.!\?])\s+");

            var gen = new Random(42);

            for (int i = 0; i < quantity; i++) {
                yield return new Record(sentences[gen.Next(sentences.Length)]);
            }
        }
    }
}

File: VB/GridControlCellTemplate/Model/Item.vb
Imports DevExpress.Mvvm
Imports System.Collections.Generic
Imports System.Text.RegularExpressions

Namespace GridControlCellTemplate.Model

    Public Class Record
        Inherits BindableBase

        Public Property Text As String
            Get
                Return GetValue(Of String)()
            End Get

            Set(ByVal value As String)
                SetValue(value)
            End Set
        End Property

        Public Property IsRead As Boolean
            Get
                Return GetValue(Of Boolean)()
            End Get

            Set(ByVal value As Boolean)
                SetValue(value)
            End Set
        End Property

        Public Sub New(ByVal text As String)
            Me.Text = text
        End Sub

        Public Shared Iterator Function GetData(ByVal quantity As Integer) As IEnumerable(Of Record)
            Dim sentences = Regex.Split(File.ReadAllText("Model/Sentences.txt"), "(?<=[\.!\?])\s+")
            Dim gen = New Random(42)
            For i As Integer = 0 To quantity - 1
                Yield New Record(sentences(gen.[Next](sentences.Length)))
            Next
        End Function
    End Class
End Namespace
